src/Compilers/Core/VBCSCompiler/ServerDispatcher.cs

```diff
@@ -147,8 +147,6 @@ public ServerDispatcher(IRequestHandler handler, IDiagnosticListener diagnosticL
             Justification = "We intentionally call GC.Collect when anticipate long period on inactivity.")]
         public void ListenAndDispatchConnections(string pipeName, TimeSpan? keepAlive, CancellationToken cancellationToken = default(CancellationToken))
         {
-            Debug.Assert(SynchronizationContext.Current == null);
```

Comment:
@jasonmalinowski It's invalid. I don't think we can assert anything about the test environment's synchronization context.